File: frontend/src/api/adminEmailLogs.ts
import client from './client'

export interface EmailLog {
  id: number
  to: string
  kind: 'password_reset' | 'provisioning' | 'announcement' | string
  status: 'sent' | 'failed' | 'skipped' | string
  error_msg: string | null
  created_at: string
}

export interface EmailLogFilters {
  kind?: string
  status?: string
  limit?: number
  offset?: number
}

export const getEmailLogs = (filters: EmailLogFilters = {}) => {
  const params = new URLSearchParams()
  if (filters.kind)   params.set('kind', filters.kind)
  if (filters.status) params.set('status', filters.status)
  if (filters.limit !== undefined)  params.set('limit', String(filters.limit))
  if (filters.offset !== undefined) params.set('offset', String(filters.offset))
  const qs = params.toString()
  return client.get<EmailLog[]>(`/api/admin/email-logs${qs ? `?${qs}` : ''}`)
}

File: backend/internal/handlers/admin_email_logs.go
package handlers

import (
	"net/http"
	"strconv"

	"idealink/internal/repository"

	"github.com/gin-gonic/gin"
)

type AdminEmailLogsHandler struct {
	repo *repository.EmailLogRepo
}

func NewAdminEmailLogsHandler(repo *repository.EmailLogRepo) *AdminEmailLogsHandler {
	return &AdminEmailLogsHandler{repo: repo}
}

// List returns the most recent email_logs rows. Query params:
//   limit  (default 50, max 200)
//   offset (default 0)
//   kind   (optional: password_reset | provisioning | announcement)
//   status (optional: sent | failed | skipped)
func (h *AdminEmailLogsHandler) List(c *gin.Context) {
	limit, _ := strconv.Atoi(c.DefaultQuery("limit", "50"))
	offset, _ := strconv.Atoi(c.DefaultQuery("offset", "0"))
	kind := c.Query("kind")
	status := c.Query("status")

	rows, err := h.repo.List(kind, status, limit, offset)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusOK, rows)
}
